pisp_app/lib/Screens/transaction_amount.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:pispapp/MockData/account.dart';
+import 'package:pispapp/Screens/dashboard.dart';
+import 'package:pispapp/Screens/transaction_details.dart';
+
+class TransactionAmount extends StatefulWidget {
```

Comment:
This could also be converted into a `StatelessWidget`.

In general, I think it is better to use `StatelessWidget` unless you need to maintain some states that can change dynamically.

Reply:
I agree. But `amount` is mutable and flutter recommends all instance fields of immutable classes to be final.

Reply:
the static analysis check wont be passed otherwise

Reply:
Oh right I didn't see the `onChanged: (String text) { amount = text; }` ... I guess you forgot to call the `setState` there?

Reply:
not really required because the textinput takes care of rendering changes. We just want to keep track of the new amount so that we can pass it down the tree.

Reply:
I see, okay got it